### PR TITLE
Share activities to user's friends in app

### PR DESCRIPTION
**Pull request by CassioManuguerra**
# PR #291 
## Overview
This PR adds an option to share activities directly in the app to your friends. It lets you choose wether you want to share outside or inside the app, and then shows you a list of your friends to select from (if you chose inside app). The PR is fully tested (>90% only mainNavGraph not tested). For now it sends the shared activity's ID to the selected friend.

https://github.com/LASTA-OUTDOOR/lasta/assets/91049597/2875b539-7cd2-4e87-9c2b-c70656ce8507

## Changes
- Added a new share options dialog
- Added a friend list to choose and share to
- Adapted correct passing of arguments to allow this feature
- Created a new function in the conversationViewModel to send a message to a certain friendID
- Adapted tests
- Created new tests

## What's next
- Modify the messages so that activities sent to users appear as clickable links to the activity's info
- Implement the feature to add a message with the activity sent (for example : Ayo Shawn, check out this cool activity, are you up to swooop it with me next Friday ?  --  Activity link here --)

### CODE REVIEW

**Comment by CassioManuguerra** [comment]
The share options :
Screenshot 2024-05-22 at 11 30 08

File: app/src/main/java/com/lastaoutdoor/lasta/ui/components/ShareSheets.kt [hunk 1]
```diff
@@ -34,7 +34,7 @@ fun shareActivity(activity: Activity, context: Context) {
   val textFromActivity =
       context.getString(R.string.share_body) +
           "\n${activity.name} \n" +
-          context.getString(R.string.share_base_url) +
```
Comment (npfrei) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/components/ShareSheets.kt:37]: Please add the URL of the current release of the app

**Review by npfrei — CHANGES_REQUESTED** [review]
Great Job ! Sharing activities is a great feature for the social aspect of the app. You just forgot to put an actual URL when sharing outside the app. Otherwise everything works great !

File: app/src/main/java/com/lastaoutdoor/lasta/ui/components/ShareSheets.kt (hunk 1 above)
Comment (CassioManuguerra) [reply app/src/main/java/com/lastaoutdoor/lasta/ui/components/ShareSheets.kt:37]: I did it ! will have to be changed though

**Review by npfrei — APPROVED** [review]
Thanks for adding my requested changes !

**Pull request merged**